controllers/volumereplication_controller.go

```diff
@@ -162,26 +174,49 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	case replicationv1alpha1.Primary:
```

Comment:
How about using the error variable above the switch case and checking it and updating the status once after the switch case. 

```
var replicationErr error 
switch instance.Spec.ImageState {
	case replicationv1alpha1.Primary:
		replicationErr = r.markVolumeAsPrimary(volumeHandle, parameters, secret)
	case replicationv1alpha1.Secondary:
		replicationErr = r.markVolumeAsSecondary(volumeHandle, parameters, secret)
	case replicationv1alpha1.Resync:
		replicationErr := r.resyncVolume(volumeHandle, parameters, secret)
	default:
		replicationErr = fmt.Errorf("unsupported image state %q", instance.Spec.ImageState)
	}


if replicationErr != nil{
   err := r.updateStatusStateAndMessage(instance, replicationv1alpha1.ReplicationFailure, replicationErr.Error())
   if err != nil{
     // log that updating status failed.
     return ctrl.Result{}, err
   }
   return ctrl.Result{}, err
}
```

Reply:
Done